node_modules/gh-auth/lib/shibboleth/strategy.js

```diff
@@ -0,0 +1,112 @@
+     * there should be a `shib-session-id` header in the request. If the user is indicating that he
+     * wants to log in with Shibboleth, no such header will be present.
+     *
+     * It's up to the front-end load-balancer (either Apache or Nginx) to *NOT* proxy these headers.
+     */
+    var sessionId = req.headers['shib-session-id'];
+    if (sessionId) {
+        /*
+         * The user is coming back from the IdP. mod_shib will pass all the user his attributes as headers.
+         *
+         * The following is a subset of the possible headers that the Shibboleth SP software proxies:
+         *  - remote_user                   :   Identifies the user with the application. Shibboleth needs to have been configured appropriately so that it knows on which attributes this value is based on
+         *  - Shib-Application-ID           :   The applicationId property derived for the request.
+         *  - Shib-Session-ID               :   The internal session key assigned to the session associated with the request.
+         *  - Shib-Identity-Provider        :   The entityID of the IdP that authenticated the user associated with the request.
+         *
+         * Depending on what profile attributes the IdP releases, other headers might be in the request.
+         * The Shibboleth SP can be configured to map certain attributes to request headers. Ideally, no mapping should happen
+         * within OAE but in the shib software. See https://wiki.shibboleth.net/confluence/display/SHIB2/NativeSPAttributeAccess
+         * for more information. As the headers are exposed as a simple JSON object on the request, we can pass them straight
+         * into the verify function, which can then take care of getting/creating the OAE user object.
+         */
+        self.verify(req, app, req.headers, function(err, user) {
+            if (err) {
+                return self.error(new Error(err.msg));
+            }
+
+            // We pass it on to passport so it can be stored in the express session object
+            return self.success(user);
+        });
+    } else {
+        /*
+         * There is no session yet. We redirect the user to Shibboleth.
+         * In case we know which IdP we want to use, we can append its entityID in the redirectUrl.
+         * For example, If we want to authentication with the Cambridge IdP
+         *      /Shibboleth.sso/Login?target=/api/auth/shibboleth/sp/callback&entityId=https://shib.raven.cam.ac.uk/shibboleth
+         *
+         * The `target` parameter in the above example is where the IdP should send the user to
+         * once he is succesfully logged in
+         */
+        var redirectUrl = util.format('/Shibboleth.sso/Login?target=%s', encodeURIComponent('/api/auth/shibboleth/sp/callback'));
+        var shibIdpEntityId = ConfigAPI.config(app.id).shibIdpEntityId;
+        if (shibIdpEntityId) {
```

Comment:
No, I was in 2 minds about this tbh. We could:
a/ require that config value
b/ don't require and use WAYF when not specified.

Thinking about it some more, we probably just want to go with option A for now.